### PR TITLE
GH-94438: Account for NULLs on evaluation stack when jumping lines.

### PR DESCRIPTION
**Pull request by markshannon**
* Issue: gh-94438

### CODE REVIEW

File: Objects/frameobject.c [hunk 1]
```diff
@@ -298,6 +313,17 @@ mark_stacks(PyCodeObject *code_obj, int len)
...
+                    next_stack = push_value(next_stack, Null);
+                    stacks[i+1] = next_stack;
+                    break;
+                case LOAD_GLOBAL:
+                    if (_Py_OPARG(code[i]) & 1) {
+                        next_stack = push_value(next_stack, Null);
+                    }
+                    next_stack = push_value(next_stack, Object);
+                    stacks[i+1] = next_stack;
+                    break;
```
Comment (Fidget-Spinner) [inline Objects/frameobject.c:328]: Do we need an entry for `LOAD_ATTR` (`LOAD_METHOD`) here?

File: Lib/test/test_sys_settrace.py [hunk 2]
```diff
@@ -2584,6 +2584,24 @@ async def test_jump_backward_over_async_listcomp_v2(output):
...
+        (
+            ( # 7
+                a
+                +
+                10
+            )
+            +
+            13
+        )
+        output.append(15)
```
Comment (iritkatriel) [inline Lib/test/test_sys_settrace.py:2604]: The  print tests the LOAD_GLOBAL case, right? Is there a test for PUSH_NULL as well?

Comment (markshannon) [reply Lib/test/test_sys_settrace.py:2604]: No, nor for `LOAD_METHOD` which I also need to add.

File: Objects/frameobject.c (hunk 1 above)
Comment (markshannon) [reply Objects/frameobject.c:328]: Yes

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**